cql3/type_json.cc

```diff
@@ -151,12 +151,16 @@ static bytes from_json_object_aux(const map_type_impl& t, const rjson::value& va
     std::map<bytes, bytes, serialized_compare> raw_map(t.get_keys_type()->as_less_comparator());
     for (auto it = value.MemberBegin(); it != value.MemberEnd(); ++it) {
         bytes value = from_json_object(*t.get_values_type(), it->value);
-        if (!t.get_keys_type()->is_compatible_with(*utf8_type)) {
+        if (!t.get_keys_type()->is_compatible_with(*ascii_type)) {
```

Comment:
Looks like `utf8_type` is compatible with `ascii_type` and not the other way around:
https://github.com/scylladb/scylladb/blob/a56a4b6226e6afd1716469e3b285c474ad8f0457/test/boost/types_test.cc#L949-L953

Reply:
i think UTF-8 is designed to be compatible with ASCII. please squash this change into the first commit. so the review can review the fixed version.

Reply:
Sorry. I didn't realize each commit would get looked at 😅. Typically I automatically squash all the commits when I merge PRs.

Reply:
An ascii string's byte representation is compatible with bytes' byte representation, but not vice versa. The bytes object 0xff is not a valid ascii string.